pkg/ingester/metrics_test.go

```diff
@@ -731,9 +739,21 @@ func populateTSDBMetrics(base float64) *prometheus.Registry {
 
 	exemplarsOutOfOrderTotal := promauto.With(r).NewCounter(prometheus.CounterOpts{
 		Name: "prometheus_tsdb_exemplar_out_of_order_exemplars_total",
-		Help: "Total number of out of order exemplar ingestion failed attempts.",
+		Help: "Total number of out-of-order exemplar ingestion failed attempts.",
 	})
 	exemplarsOutOfOrderTotal.Add(3)
 
+	outOfOrderSamplesAppendedTotal := promauto.With(r).NewCounter(prometheus.CounterOpts{
+		Name: "prometheus_tsdb_head_out_of_order_samples_appended_total",
+		Help: "Total number of appended out-of-order samples.",
+	})
+	outOfOrderSamplesAppendedTotal.Add(3)
+
+	tooOldSamplesTotal := promauto.With(r).NewCounter(prometheus.CounterOpts{
+		Name: "prometheus_tsdb_too_old_samples_total",
```

Comment:
I'm not super convinced about the metric name. I think it would be more clear a `prometheus_tsdb_head_out_of_order_samples_append_failed_total` with a `reason` label.